### PR TITLE
[ICU-648] Fix download tooltip

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Move tooltip from filename to download icon

Others: 
- Refactor `FileAttachment` component
- Add `FileThumbnail` component

@esethna Could you please check if the tooltip is positioned as you would expect? Thanks!

screen shot 2018-02-01 at 6 15 33 am

Update (Feb 2):
screen shot 2018-02-02 at 7 45 48 pm
screen shot 2018-02-02 at 7 45 11 pm

#### Ticket Link
This PR addressed item 1 of:
- Jira ticket: [ICU-648](https://mattermost.atlassian.net/browse/ICU-648)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes

### CODE REVIEW

File: components/file_attachment.jsx [hunk 1]
```diff
@@ -158,26 +112,36 @@ export default class FileAttachment extends React.PureComponent {
...
-                    {downloadButton}
+                    {canDownload &&
+                    <FilenameOverlay
+                        fileInfo={fileInfo}
+                        compactDisplay={compactDisplay}
+                        canDownload={canDownload}
+                        iconClass={'post-image__download'}
+                        index={index}
+                    >
+                        <DownloadIcon/>
```
Comment (saturninoabril) [inline components/file_attachment.jsx]: This is the actual fix. Others are just refactor.

**Comment by saturninoabril** [comment]
@esethna Thanks for the review. I have addressed both 1 & 2. See screenshots below.

screen shot 2018-02-02 at 7 45 48 pm
screen shot 2018-02-02 at 7 45 11 pm

File: components/file_attachment.jsx [hunk 2]
```diff
@@ -40,7 +47,7 @@ export default class FileAttachment extends React.PureComponent {
         super(props);
 
         this.state = {
-            loaded: Utils.getFileType(props.fileInfo.extension) !== 'image'
+            loaded: getFileType(props.fileInfo.extension) !== 'image'
```
Comment (mkraft) [inline components/file_attachment.jsx]: Not part of this PR, I know, but it drives me crazy that `getFileType` doesn't return constant-defined strings that could be compared against. Would be great if someone refactored that...

Comment (saturninoabril) [reply components/file_attachment.jsx]: Yeah, that makes sense, I'll submit separate PR to refactor that.

Comment (saturninoabril) [reply components/file_attachment.jsx]: PR submitted https://github.com/mattermost/mattermost-webapp/pull/701

**Comment by saturninoabril** [comment]
Rebased to latest and fixed merge conflicts
